metrics/opentelemetry/engine.go

```diff
@@ -0,0 +1,678 @@
+			SuccessNoops    metric.Int64Counter     `description:"Count of module successful noops labeled by stage name" unit:"1"`
+			SuccessUpdates  metric.Int64Counter     `description:"Count of module successful updates labeled by stage name" unit:"1"`
+			SuccessRejects  metric.Int64Counter     `description:"Count of module successful rejects labeled by stage name" unit:"1"`
+			ExecutionErrors metric.Int64Counter     `description:"Count of module execution errors labeled by stage name" unit:"1"`
+			Timeouts        metric.Int64Counter     `description:"Count of module timeouts labeled by stage name" unit:"1"`
+		}
+	}
+	PbsMetricsEngine struct {
+		*PbsMetrics
+		MetricsDisabled *config.DisabledMetrics
+	}
+)
+
+const (
+	markupDeliveryAdm  = "adm"
+	markupDeliveryNurl = "nurl"
+)
+
+const (
+	accountLabel         attribute.Key = "account"
+	actionLabel          attribute.Key = "action"
+	adapterErrorLabel    attribute.Key = "adapter_error"
+	adapterLabel         attribute.Key = "adapter"
+	bidTypeLabel         attribute.Key = "bid_type"
+	cacheResultLabel     attribute.Key = "cache_result"
+	connectionErrorLabel attribute.Key = "connection_error"
+	cookieLabel          attribute.Key = "cookie"
+	hasBidsLabel         attribute.Key = "has_bids"
+	isAudioLabel         attribute.Key = "audio"
+	isBannerLabel        attribute.Key = "banner"
+	isNativeLabel        attribute.Key = "native"
+	isVideoLabel         attribute.Key = "video"
+	markupDeliveryLabel  attribute.Key = "delivery"
+	optOutLabel          attribute.Key = "opt_out"
+	overheadTypeLabel    attribute.Key = "overhead_type"
+	privacyBlockedLabel  attribute.Key = "privacy_blocked"
+	requestStatusLabel   attribute.Key = "request_status"
+	requestTypeLabel     attribute.Key = "request_type"
+	stageLabel           attribute.Key = "stage"
+	statusLabel          attribute.Key = "status"
+	successLabel         attribute.Key = "success"
+	syncerLabel          attribute.Key = "syncer"
+	versionLabel         attribute.Key = "version"
+
+	storedDataFetchTypeLabel attribute.Key = "stored_data_fetch_type"
+	storedDataErrorLabel     attribute.Key = "stored_data_error"
+
+	requestSuccessLabel attribute.Key = "requestAcceptedLabel"
+	requestRejectLabel  attribute.Key = "requestRejectedLabel"
+
+	sourceLabel   attribute.Key = "source"
+	sourceRequest attribute.Key = "request"
+
+	moduleLabel attribute.Key = "module"
+)
+
+const (
+	requestSuccessful = "ok"
+	requestFailed     = "failed"
+)
+
+var (
+	_ metrics.MetricsEngine = &PbsMetricsEngine{}
+
+	Meter = otel.Meter("github.com/prebid/prebid-server/v3")
+)
+
+// NewEngine creates a new OpenTelemetry engine with the given prefix
+func NewEngine(prefix string, disabledMetrics *config.DisabledMetrics) (*PbsMetricsEngine, error) {
+	ret := &PbsMetricsEngine{
+		PbsMetrics:      &PbsMetrics{},
+		MetricsDisabled: disabledMetrics,
+	}
+	if err := InitMetrics(Meter, ret.PbsMetrics, prefix); err != nil {
+		return nil, err
+	}
+	return ret, nil
+}
```

Comment:
How are OpenTelemetry metrics published?

Reply:
They are published by the configuration of the sdk - this only hooks up the otel interfaces and does not provide configuration of the sdk. If a host, using prebid as a library/module, configures otel, then these metrics will go wherever it specifies.